R/data.table.R

```diff
@@ -1016,6 +1018,7 @@ replace_dot_alias = function(e) {
           if (anyNA(.SDcols))
             stop(".SDcols missing at the following indices: ", brackify(which(is.na(.SDcols))))
           if (is.logical(.SDcols)) {
+            if (length(.SDcols)!=length(x)) warning("logical vector in .SDcols should have equal length to number of columns in input table")
```

Comment:
We usually try and be even more helpful in error messages by including the values of `length(x)` and `length(.SDcols)` in the message.